pyqtgraph/functions.py

```diff
@@ -1794,9 +1799,66 @@ def arrayToQPath(x, y, connect='all'):
 
     ds = QtCore.QDataStream(buf)
     ds >> path
-
     return path
 
+def arrayToQPolygonF(x, y):
+    """
+    Utility function to convert two 1D-NumPy arrays representing curve data
+    (X-axis, Y-axis data) into a single open polygon (QtGui.PolygonF) object.
+    
+    Thanks to PythonQwt for making this code available
+    
+    License/copyright: MIT License © Pierre Raybaut 2020.
+
+    Parameters
+    ----------
+    x : np.array
+        x-axis coordinates for data to be plotted, must have np.float64 dtype 
+        and have ndim of 1
+    y : np.array
+        y-axis coordinates for data to be plotted, must have np.float64 dtype,
+        ndim of 1 and be the same length as x-axis
+    
+    Returns
+    -------
+    QPolygonF
+        Open QPolygonF object that represents the path looking to be plotted
+    
+    Raises
+    ------
+    ValueError
+        When xdata or ydata does not meet the required criteria
+    """
+    if not (
+        x.size == y.size == x.shape[0] == y.shape[0]
+        and np.issubdtype(x.dtype, np.double)
+        and np.issubdtype(y.dtype, np.double)
+    ):
+        raise ValueError("Arguments must be 1D, float64 NumPy arrays with same size")
+    size = x.size
+    dtype = x.dtype
+    if QtVersion.startswith("5"):
+        polyline = QtGui.QPolygonF(size)
+    else:
+        polyline = QtGui.QPolygonF()
+        if QT_LIB == "PySide6":
+            polyline.resize(size)
+        else:
+            polyline.fill(QtCore.QPointF(), size)
+    nbytes = 2 * size * np.finfo(dtype).dtype.itemsize
+    if QT_LIB == "PySide2":
+        buffer = Qt.shiboken2.VoidPtr(polyline.data(), nbytes, True)
+    elif QT_LIB == "PySide6":
+        buffer = Qt.shiboken6.VoidPtr(polyline.data(), nbytes, True)
+    else:
+        buffer = polyline.data()
+        buffer.setsize(nbytes)
+
+    memory = np.frombuffer(buffer, np.double)
```

Comment:
Let numpy do the calculation of memory offsets
```python
memory = np.frombuffer(buffer, np.double).reshape((-1, 2))
memory[:, 0] = x
memory[:, 1] = y
```

Reply:
yes! that's a lot cleaner!  Implemented that version now.